test/deploy/rackhd_ha_install.py

```diff
@@ -0,0 +1,109 @@
+'''
+Copyright 2017, Dell EMC, Inc.
+
+This script prepares the VMs for an HA clustered configuration
+This is for setting up a cluster to do HA development and testing.
+
+
+This script performs the following functions:
+    - Configures the control network for each node in the cluster
+    - installs the DHCP configuration
+    - sets the /etc/hosts config
+    - sets the hostname
+
+usage:
+    python run_tests.py -stack <stack ID> -test deploy/rackhd_ha_install.py -numvms <num>
+'''
+
+import os
+import sys
+import subprocess
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/common")
+import fit_common
+
+ifslist = []   # array of valid eth ports
+numvms = int(fit_common.fitargs()['numvms'])
+
+
+class rackhd_ha_install(fit_common.unittest.TestCase):
+
+    def test01_install_network_config(self):
+        for vmnum in range(1, numvms + 1):
+
+            # collect nic names
+            getifs = fit_common.remote_shell("ifconfig -s -a |tail -n +2 |grep -v -e Iface -e lo -e docker", vmnum=vmnum)
+            # clean out login stuff
+            splitifs = getifs['stdout'].split('\n')
+            for item in splitifs:
+                if "assword" not in item and item.split(" ")[0]:
+                    ifslist.append(item.split(" ")[0])
+
+            control_ip = '172.31.128.{}'.format(vmnum)
+
+            # install control network config
+            control_cfg = open('control.cfg', 'w')
+            control_cfg.write('auto ' + ifslist[1] + '\n'
+                              'iface ' + ifslist[1] + ' inet static\n'
+                              'address ' + control_ip + '\n'
+                              'netmask 255.255.252.0\n')
+            control_cfg.close()
+            # copy file to ORA
+            fit_common.scp_file_to_ora('control.cfg', vmnum)
+            self.assertEqual(fit_common.remote_shell('cp control.cfg /etc/network/interfaces.d/', vmnum=vmnum)['exitcode'],
+                             0, "Control network config failure.")
+            os.remove('control.cfg')
+            # startup NIC
+            fit_common.remote_shell('ip addr add ' + control_ip + '/22 dev ' + ifslist[1], vmnum=vmnum)
+            fit_common.remote_shell('ip link set ' + ifslist[1] + ' up', vmnum=vmnum)
+            self.assertEqual(fit_common.remote_shell('ping -c 1 -w 5 ' + control_ip, vmnum=vmnum)['exitcode'],
+                             0, 'Control NIC failure.')
+
+    def test02_install_dhcp_config(self):
```

Comment:
Suggestions: For a deploy script, you could use nosedep and @depends if you want some of the tests to be skipped if the previous one failed.